--- Makefile.orig	2007-07-29 19:20:34.000000000 +0200
+++ Makefile	2007-07-29 19:17:23.000000000 +0200
@@ -473,5 +473,8 @@
 		"$(SGPDATADIR)"/Data/TILECACHE/*.[Jj][Ss][Dd] \
 		"$(SGPDATADIR)"/Data/TILECACHE/*.[Ss][Tt][Ii]; \
 	do \
-		mv "$$i" "`dirname "$$i"`/`basename "$$i" | LANG=C tr '[A-Z]' '[a-z]'`"; \
+		l="`basename "$$i" | LANG=C tr '[A-Z]' '[a-z]'`"; \
+		if test "`basename "$$i"`" != "$$l"; then  \
+			mv -u "$$i" "`dirname "$$i"`"/"$$l"; \
+		fi; \
 	done
