diff -Naur point2play.orig/usr/lib/transgaming_point2play/setup_wizard_gui.py point2play/usr/lib/transgaming_point2play/setup_wizard_gui.py
--- point2play.orig/usr/lib/transgaming_point2play/setup_wizard_gui.py	2005-06-02 09:18:04.000000000 -0700
+++ point2play/usr/lib/transgaming_point2play/setup_wizard_gui.py	2005-06-28 14:08:46.338824184 -0700
@@ -111,7 +111,8 @@
         # not all the data widgets will be autodetected. It's OK to leave some out.
         system_info = system_detection.make_detected_system_info_dictionary()
         for w in system_detection.system_data_widgets_list:
-            self.widgets[w].set_text(system_info[w])
+            if system_info[w]:
+                self.widgets[w].set_text(system_info[w])
 
     def detect_system_info_cb(self, button):
         self.detect_system_info()
