diff -u -r vmware-vix-distrib-orig/bin/vmware-uninstall-vix.pl vmware-vix-distrib/bin/vmware-uninstall-vix.pl
--- vmware-vix-distrib-orig/bin/vmware-uninstall-vix.pl	2007-10-08 10:18:10.000000000 -0500
+++ vmware-vix-distrib/bin/vmware-uninstall-vix.pl	2007-11-28 23:21:18.000000000 -0600
@@ -7,11 +7,15 @@
 
 use strict;
 
+$| = 1;
+
 # BEGINNING_OF_UTIL_DOT_PL
 #!/usr/bin/perl
 
 use strict;
 
+$| = 1;
+
 # Moved out of config.pl to support $gOption in spacechk_answer
 my %gOption;
 # Moved from various scripts that include util.pl
@@ -2737,45 +2741,11 @@
     return $initscriptsdir;
   }
 
-  # The "SuSE version >= 7.1" way
-  $initdir = '/etc/init.d';
-  if (check_answer_initdirpath($initdir, 'default') eq '') {
-    # The "SuSE version < 7.1" way
-    $initdir = '/sbin/init.d';
-    if (check_answer_initdirpath($initdir, 'default') eq '') {
-      # The "RedHat" way
-      $initdir = '/etc/rc.d';
-      if (check_answer_initdirpath($initdir, 'default') eq '') {
-        # The "Debian" way
-        $initdir = '/etc';
-        if (check_answer_initdirpath($initdir, 'default') eq '') {
-          $initdir = '';
-        }
-      }
-    }
-  }
-  $answer = get_persistent_answer('What is the directory that contains the init'
-                                  .' directories (rc0.d/ to rc6.d/)?'
-                                  , 'INITDIR', 'initdirpath', $initdir);
-
-  # The usual way
-  $initscriptsdir = $answer . '/init.d';
-  if ( $answer =~ m/init.d/ ) {
-    # if the string contains init.d, do not default to containing init.d,
-    # instead just default to the initdir as the initscripstdir
-    $initscriptsdir = $answer;
-  }
-
-  if (check_answer_initscriptsdirpath($initscriptsdir, 'default') eq '') {
-    # The "SuSE version >= 7.1" way
-    $initscriptsdir = $answer;
-    if (check_answer_initscriptsdirpath($initscriptsdir, 'default') eq '') {
-      $initscriptsdir = '';
-    }
-  }
-  $answer = get_persistent_answer('What is the directory that contains the init'
-                                  .' scripts?', 'INITSCRIPTSDIR'
-                                  , 'initscriptsdirpath', $initscriptsdir);
+  # The "Fake smgl" way
+  $initscriptsdir = $answer = $initdir = '/tmp/vmware_rc.d';
+  db_add_answer('INITDIR', $answer);
+  db_add_answer('INITSCRIPTSDIR', $answer);
+
   return $answer;
 }
 
diff -u -r vmware-vix-distrib-orig/installer/services.sh vmware-vix-distrib/installer/services.sh
--- vmware-vix-distrib-orig/installer/services.sh	2007-10-08 10:18:11.000000000 -0500
+++ vmware-vix-distrib/installer/services.sh	2007-11-28 23:22:22.000000000 -0600
@@ -26,6 +26,11 @@
 # BEGINNING_OF_UTIL_DOT_SH
 #!/bin/sh
 
+function echo ()
+{
+       builtin echo "$@"
+}
+
 # A few utility functions used by our shell scripts. These are patched in 
 # during make.
 
Only in vmware-vix-distrib-orig: vmware-install.pl
