diff -Naur acroread-5.0.10.orig/installers/INSTALL acroread-5.0.10/installers/INSTALL
--- acroread-5.0.10.orig/installers/INSTALL	2005-01-05 20:02:37.558588040 -0800
+++ acroread-5.0.10/installers/INSTALL	2005-01-05 20:02:52.747279008 -0800
@@ -344,7 +344,6 @@
   SearchPlatformTar="$3"
   CustomPlatformTar="$4"
 
-  OutputLicense "$ReadLicense"
   ReadLicense=""
 
   PrintRequiredFree "$ReadTar" "$SearchTar" "$CustomTar" \
