--- glext.h~	2006-11-08 22:09:16.000000000 -0800
+++ glext.h	2006-11-08 22:09:16.000000000 -0800
@@ -6938,7 +6938,6 @@
 extern void APIENTRY glGetVertexAttribIivEXT (GLuint index, GLenum pname, GLint *params);
 extern void APIENTRY glGetVertexAttribIuivEXT (GLuint index, GLenum pname, GLuint *params);
 #endif /* GL_GLEXT_PROTOTYPES */
-typedef void (APIENTRY * PFNGLGETUNIFORMUIVEXTPROC) (GLuint program, GLint location, GLuint *params);
 typedef void (APIENTRY * PFNGLBINDFRAGDATALOCATIONEXTPROC) (GLuint program, GLuint color, const GLchar *name);
 typedef GLint (APIENTRY * PFNGLGETFRAGDATALOCATIONEXTPROC) (GLuint program, const GLchar *name);
 typedef void (APIENTRY * PFNGLUNIFORM1UIEXTPROC) (GLint location, GLuint v0);
