--- winex-2-2-1/files/drive.c	2002-10-03 10:29:14.000000000 -0500
+++ drive.c	2003-07-05 02:43:41.000000000 -0500
@@ -307,8 +307,8 @@
 	drive->dev = drive_stat_buffer.st_dev;
 	drive->ino = drive_stat_buffer.st_ino;
 	DRIVE_AddCDReg(name[6]);
-	TRACE("Found from fstab - %s: path=%s type=%s label='%s' serial=%08lx 
-			flags=%08x dev=%x ino=%x\n",   name, drive->root, DRIVE_Types[drive->type],
+	TRACE("Found from fstab - %s: path=%s type=%s label='%s' serial=%08lx \nflags=%08x dev=%x ino=%x\n", 
+			name, drive->root, DRIVE_Types[drive->type],
 		        drive->label_conf, drive->serial_conf, drive->flags,
 			(int)drive->dev, (int)drive->ino );
 }
