diff -u -r vmware-distrib.orig/bin/vmware-config.pl vmware-distrib/bin/vmware-config.pl
--- vmware-distrib.orig/bin/vmware-config.pl	2006-12-02 13:39:25.000000000 -0600
+++ vmware-distrib/bin/vmware-config.pl	2006-12-02 13:44:24.000000000 -0600
@@ -7,6 +7,8 @@
 
 use strict;
 
+$| = 1;
+
 # Use Config module to update VMware host-wide configuration file
 # BEGINNING_OF_CONFIG_DOT_PM
 #!/usr/bin/perl
@@ -21,6 +23,9 @@
 ###
 
 use strict;
+
+$| = 1;
+
 package VMware::Config;
 
 my %PREF;
@@ -1532,6 +1537,8 @@
 
 use strict;
 
+$| = 1;
+
 # Create a temporary directory
 #
 # They are a lot of small utility programs to create temporary files in a
diff -u -r vmware-distrib.orig/bin/vmware-uninstall.pl vmware-distrib/bin/vmware-uninstall.pl
--- vmware-distrib.orig/bin/vmware-uninstall.pl	2006-12-02 13:39:25.000000000 -0600
+++ vmware-distrib/bin/vmware-uninstall.pl	2006-12-02 13:38:03.000000000 -0600
@@ -7,6 +7,8 @@
 
 use strict;
 
+$| = 1;
+
 # Use Config module to update VMware host-wide configuration file
 # BEGINNING_OF_CONFIG_DOT_PM
 # END_OF_CONFIG_DOT_PM
@@ -2054,39 +2056,11 @@
     return $initscriptsdir;
   }
 
-  # The "SuSE version >= 7.1" way
-  $initdir = '/etc/init.d';
-  if (check_answer_initdirpath($initdir, 'default') eq '') {
-    # The "SuSE version < 7.1" way
-    $initdir = '/sbin/init.d';
-    if (check_answer_initdirpath($initdir, 'default') eq '') {
-      # The "RedHat" way
-      $initdir = '/etc/rc.d';
-      if (check_answer_initdirpath($initdir, 'default') eq '') {
-        # The "Debian" way
-        $initdir = '/etc';
-        if (check_answer_initdirpath($initdir, 'default') eq '') {
-          $initdir = '';
-        }
-      }
-    }
-  }
-  $answer = get_persistent_answer('What is the directory that contains the init'
-                                  .' directories (rc0.d/ to rc6.d/)?'
-                                  , 'INITDIR', 'initdirpath', $initdir);
-
-  # The usual way
-  $initscriptsdir = $answer . '/init.d';
-  if (check_answer_initscriptsdirpath($initscriptsdir, 'default') eq '') {
-    # The "SuSE version >= 7.1" way
-    $initscriptsdir = $answer;
-    if (check_answer_initscriptsdirpath($initscriptsdir, 'default') eq '') {
-      $initscriptsdir = '';
-    }
-  }
-  $answer = get_persistent_answer('What is the directory that contains the init'
-                                  .' scripts?', 'INITSCRIPTSDIR'
-                                  , 'initscriptsdirpath', $initscriptsdir);
+  # The "Fake smgl" way
+  $initscriptsdir = $answer = $initdir = '/tmp/vmware_rc.d';
+  db_add_answer('INITDIR', $answer);
+  db_add_answer('INITSCRIPTSDIR', $answer);
+
   return $answer;
 }
 
diff -u -r vmware-distrib.orig/installer/services.sh vmware-distrib/installer/services.sh
--- vmware-distrib.orig/installer/services.sh	2006-12-02 13:39:25.000000000 -0600
+++ vmware-distrib/installer/services.sh	2006-12-02 13:45:55.000000000 -0600
@@ -26,6 +26,11 @@
 # BEGINNING_OF_UTIL_DOT_SH
 #!/bin/sh
 
+function echo ()
+{
+       builtin echo "$@"
+}
+
 # A few utility functions used by our shell scripts. These are patched in during
 # make.
 

