From 34220f2ff041303594b0a5c8c112ca2054550739 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Fri, 5 Aug 2016 10:41:23 +0200
Subject: [PATCH] fix multiple definition of radix_tree_empty

---
 kernel/nvidia-uvm/uvm_linux.h | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/kernel/nvidia-uvm/uvm_linux.h b/kernel/nvidia-uvm/uvm_linux.h
index 93dc997..bdeddf7 100644
--- a/kernel/nvidia-uvm/uvm_linux.h
+++ b/kernel/nvidia-uvm/uvm_linux.h
@@ -547,11 +547,5 @@ static void uvm_init_radix_tree_preloadable(struct radix_tree_root *tree)
     INIT_RADIX_TREE(tree, GFP_NOWAIT);
 }
 
-static bool radix_tree_empty(struct radix_tree_root *tree)
-{
-    void *dummy;
-    return radix_tree_gang_lookup(tree, &dummy, 0, 1) == 0;
-}
-
 #endif // _UVM_LINUX_H
 
-- 
2.9.2

