From 5e145a34ec70c53c46ccdc2c9f821a1dffacea4f Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Fri, 5 Aug 2016 10:24:54 +0200
Subject: [PATCH] fix multiple definition of radix_tree_empty

---
 kernel/nvidia-uvm/uvm_linux.h | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/kernel/nvidia-uvm/uvm_linux.h b/kernel/nvidia-uvm/uvm_linux.h
index e150687..796bdd2 100644
--- a/kernel/nvidia-uvm/uvm_linux.h
+++ b/kernel/nvidia-uvm/uvm_linux.h
@@ -554,12 +554,6 @@ static void uvm_init_radix_tree_preloadable(struct radix_tree_root *tree)
     INIT_RADIX_TREE(tree, GFP_NOWAIT);
 }
 
-static bool radix_tree_empty(struct radix_tree_root *tree)
-{
-    void *dummy;
-    return radix_tree_gang_lookup(tree, &dummy, 0, 1) == 0;
-}
-
 
 #if !defined(NV_USLEEP_RANGE_PRESENT)
 static void __sched usleep_range(unsigned long min, unsigned long max)
-- 
2.9.2

