From d57c6bf7799dd815b6d9b0abbe9e86c2cd0ead21 Mon Sep 17 00:00:00 2001
From: Florian Franzmann <siflfran@hawo.stw.uni-erlangen.de>
Date: Mon, 20 Jan 2014 20:23:58 +0100
Subject: [PATCH] fix error due to changed acpi api

---
 kernel/nv-acpi.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/nv-acpi.c b/kernel/nv-acpi.c
index b0138cd..6c55144 100644
--- a/kernel/nv-acpi.c
+++ b/kernel/nv-acpi.c
@@ -303,7 +303,9 @@ static int nv_acpi_remove(struct acpi_device *device, int type)
 
     if (pNvAcpiObject->notify_handler_installed)
     {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 13, 0)
         NV_ACPI_OS_WAIT_EVENTS_COMPLETE();
+#endif
 
         // remove event notifier
         status = acpi_remove_notify_handler(device->handle, ACPI_DEVICE_NOTIFY, nv_acpi_event);
-- 
1.8.5.3

