diff -Naur NVIDIA-Linux-x86-1.0-7167-pkg1.orig/usr/include/GL/glx.h NVIDIA-Linux-x86-1.0-7167-pkg1/usr/include/GL/glx.h
--- NVIDIA-Linux-x86-1.0-7167-pkg1.orig/usr/include/GL/glx.h	2005-02-25 09:37:02.000000000 -0800
+++ NVIDIA-Linux-x86-1.0-7167-pkg1/usr/include/GL/glx.h	2005-03-12 15:13:09.184803512 -0800
@@ -37,6 +37,7 @@
 typedef XID GLXPixmap;
 typedef XID GLXDrawable;
 typedef XID GLXPbuffer;
+typedef XID GLXPbufferSGIX;
 typedef XID GLXWindow;
 typedef XID GLXFBConfigID;
 
